wcomponents-core/src/test/java/com/github/bordertech/wcomponents/servlet/ServletUtilDeviceType_Test.java

```diff
@@ -0,0 +1,77 @@
+package com.github.bordertech.wcomponents.servlet;
+
+import com.github.bordertech.wcomponents.util.DeviceType;
+import com.github.bordertech.wcomponents.util.mock.servlet.MockHttpServletRequest;
+import java.util.Arrays;
+import java.util.Collection;
+import junit.framework.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Parameterized test for device types in {@link ServletUtil}.
+ *
+ * @author Jonathan Austin
+ * @since 1.2.5
+ */
+@RunWith(Parameterized.class)
+public class ServletUtilDeviceType_Test {
+
+	private static final String MOBILE_SAFARI = "Mozilla/5.0 (iPad; CPU OS 9_3_3 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13G34 Safari/601.1";
+
+	private static final String CHROME_WIN_10 = "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36";
+
+	private static final String CHROME_MOBILE = "Mozilla/5.0 (Linux; <Android Version>; <Build Tag etc.>) AppleWebKit/<WebKit Rev> (KHTML, like Gecko) Chrome/<Chrome Rev> Mobile Safari/<WebKit Rev>";

```

Comment:
The parts of the user agent strings in angle bracket are, indeed, templates. They are not _really_ important for the purposes of these tests. If you want to replace them with some actual common user agent strings see http://www.useragentstring.com/pages/useragentstring.php